lib/db.js

```diff
@@ -637,25 +714,29 @@ Db.prototype.dropCollection = function(name, options, callback) {
  * @param {Db~resultCallback} [callback] The results callback
  * @return {Promise} returns Promise if no callback passed
  */
+const dropDatabaseSchema = {
+  w: { type: ['number', 'string'] },
```

Comment:
`w`, `wtimeout`, `j` are not documented above

Reply:
Should I be validating `w`, `wtimeout`, `j`, and `writeConcern`, or just `writeConcern`?

Reply:
I think in this case you should only be validating the documented values. Does the method actually look for these values?

Reply:
`applyWriteConcern`, used in `dropDatabase`, looks for `w`, `wtimeout`, `j`, and `fsync`.

Reply:
cool. This reminds me that `applyWriteConcern` needs to take into account a `writeConcern` property as well! Also it seems like it needs to be more defensive, changing:
```
  if (
    coll &&
    (coll.writeConcern.w != null || coll.writeConcern.j != null || coll.writeConcern.fsync != null)
  ) {
```

to:
```
  if (
    coll && coll.writeConcern &&
    (coll.writeConcern.w != null || coll.writeConcern.j != null || coll.writeConcern.fsync != null)
  ) {
```